src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -352,6 +352,41 @@ void ClusterResourceScheduler::AddLocalResource(const std::string &resource_name
   }
 }
 
+bool ClusterResourceScheduler::IsAvailableResourceEmpty(
+    const std::string &resource_name) {
+  auto it = nodes_.find(local_node_id_);
+  if (it == nodes_.end()) {
+    RAY_LOG(WARNING) << "Can't find local node:[" << local_node_id_
+                     << "] when check local available resource.";
+    return true;
+  }
+
+  int idx = -1;
+  if (resource_name == ray::kCPU_ResourceLabel) {
+    idx = (int)CPU;
+  } else if (resource_name == ray::kGPU_ResourceLabel) {
+    idx = (int)GPU;
+  } else if (resource_name == ray::kTPU_ResourceLabel) {
+    idx = (int)TPU;
+  } else if (resource_name == ray::kMemory_ResourceLabel) {
+    idx = (int)MEM;
+  };
+
+  auto local_view = it->second.GetMutableLocalView();
+  if (idx != -1) {
+    return local_view->predefined_resources[idx].available <= 0;
+  } else {
```

Comment:
We don't need else here (so that we can remove nested blocks)

Reply:
em.... sorry, I didn't get what's your mean, the code statement in `else` is for check whether customer resource available is empty.

Reply:
```cpp
auto local_view = it->second.GetMutableLocalView();
  if (idx != -1) {
    return local_view->predefined_resources[idx].available <= 0;
  }
  string_to_int_map_.Insert(resource_name);
  int64_t resource_id = string_to_int_map_.Get(resource_name);
  auto itr = local_view->custom_resources.find(resource_id);
  if (itr != local_view->custom_resources.end()) {
    return itr->second.available <= 0;
  } else {
    return true;
  }
}
```

Reply:
oh, I see, thank u~ ❤️